### ISSUE TITLE
Cannot add internal links in Bard field

### ISSUE THREAD

**Issue by boosteffect** [statamic/v2-hub#2376]
**Describe the bug**
We're trying to add internal links using the link button of the popup toolbar within Bard. Typing the title of an internal page into the link searchbox never finds the page and the link cannot be added. More details are in this discussion [https://statamic.com/forum/3917-internal-bard-links](url)

**To Reproduce**
1. In the control panel, open any page with a Bard field to edit the content.
2. Select some text in the Bard field to open the popup toolbar and click the Add Link toolbar button.
3. Enter the title of another internal page into the "Type a URL or search for an item" box.
4. The internal page is not found. No results are ever displayed from the type ahead search. The link cannot be added.

**Expected behavior**
Typing the title of an internal page in the Bard "Add Link" field should display a list of matches and the user should be able to select the internal page to add a link.

**Screenshots**
None

**Environment details (please complete the following information):**
 - Statamic Version: 2.11.10
 - Fresh Install
 - OS: Ubuntu 18.04 x64 
 - Browser: Chrome
 - Web Server: Nginx
 - PHP Version: 7.3
 - Addons installed:  FontAwesome - Typeahead search for FontAwesome icons

**Additional context**
I think the problem is in the LinkToolbar vue component. I stepped through the code and see the suggestions list of internal pages getting loaded ok. However the **_getLinkId_** method always returns null. So the computed property **_internalLink_** that would search the list of internal pages never gets called that I can tell.

**Comment by andyra** [comment]
<img width="654" alt="example" src="https://user-images.githubusercontent.com/784440/55568562-89165900-56c5-11e9-8944-8008f026998c.png">
<img width="575" alt="console-error" src="https://user-images.githubusercontent.com/784440/55568570-8ddb0d00-56c5-11e9-94d2-42b0aa3bbef8.png">

**Comment by jasonvarga** [comment]
There should be an AJAX request on the page for /cp/addons/suggest/suggestions with a list of urls being returned. Can you share what's in the response?

**Comment by andyra** [comment]
@jasonvarga I'm assuming form the URL you mentioned that you're talking about the [Suggest Addon](https://docs.statamic.com/fieldtypes/suggest), which I don't have installed. This problem is specific to the Bard field type.

**Comment by boosteffect** [comment]
I went to re-test so I could check that AJAX request and now Add Link is working fine for me. Not sure what fixed this in my environment, I haven't updated my Statamic version or made any other changes to the system. I did add the Spock addon, but I wouldn't think that is related. 

Could the issue have been related to the cache or stache needing to be refreshed?

**Comment by andyra** [comment]
I ran both `php please clear:cache` and `php please clear:stache`, and neither of them fixed the problem.

**Comment by boosteffect** [comment]
@andyra Yeah, I was just guessing on the possible cache/stache refresh fixing it on my end. I haven't made any other changes on my site other than adding Spock and some content changes.

I don't have the Suggest add-on installed either, but I can see the Ajax call to /cp/addons/suggest/suggestions that @jasonvarga mentioned being made and returning results. I think the Suggest functionality is built into Statamic core now.

**Comment by andyra** [comment]
@boosteffect Oh, I get it! I didn't realize the Suggest addon was in Statamic core now.

When you say "...I can see the Ajax call to /cp/addons/suggest/suggestions", where are you seeing that Ajax call? The error I get in the console when trying to use the Bard link functionality (screenshot above) doesn't mention anything about that, and trying to visit that URL from the Statamic control panel returns a 404.

**Comment by boosteffect** [comment]
@andyra Look under the Network tab in Chrome developer tools. Try to add a link in a Bard field and you should see a "suggestions" item in the list of network activity. This is the the AJAX call. 

![Suggestions Post](https://user-images.githubusercontent.com/35601682/57135787-a9930b00-6d78-11e9-8284-a3139674a2cd.png)

**Comment by andyra** [comment]
@boosteffect Thanks! OK, I'm seeing a couple of `suggestions` requests, each of which has what looks like relevant data that _should_ be showing up in the Bard suggestions:

<img width="664" alt="Screen Shot 2019-05-03 at 10 27 43 AM" src="https://user-images.githubusercontent.com/784440/57147942-457b4180-6d8e-11e9-8969-38b76ff2cca2.png">

**Comment by andyra** [comment]
@boosteffect @jasonvarga Is there anything else I can do to define this bug or otherwise help to move the solution forward? I'm still not able to get Bard to return any links on the latest version of Statmic (2.11.13).

To summarize:
- I'm getting a **Typeahead vue error** in the console when searching for an entry in Bard's Link dialog ([see above post for screenshot](https://github.com/statamic/v2-hub/issues/2376#issuecomment-479948865))
- I'm getting plenty of suggestion results in the Newtork tab of the Chrome inspector, but they're not making it to the UI.
- This happens in Safari, Firefox, and Chrome

**Comment by thestepafter** [comment]
I am encountering the same issue on Statamic version 2.11.19. When toggling allow internal links to true in the Brad fieldset this error occurs when trying to save. (The internal pages list also does not show.)

```
Typeahead.vue:128 Uncaught TypeError: Cannot read property '-1' of undefined
    at Typeahead.hit (Typeahead.vue:128)
    at VueComponent.<anonymous> (vue.common.js:216)
    at VueComponent.hit (Input.vue:39)
    at HTMLInputElement.<anonymous> (vue.common.js:216)
    at HTMLInputElement.<anonymous> (vue.common.js:5267)
    at HTMLInputElement.<anonymous> (vue.common.js:5252)
```

What was the resolution to this issue?

**Comment by thestepafter** [comment]
This was resolved by removing a .html entry file in collections when by default the site is using .md files. Not sure why every couple of months we get a .html file that is named the same as one of our existing .md files. That is most likely another issue. Also, just to note we have about 1300 collection entries and after the .html file was removed all is working as expected.

**Comment by andyra** [comment]
This still continues to be a problem for me. I'm on the lastest Statamic version (2.11.19) and I don't have any .html files anywhere in my content folder—everything is .md.

**Comment by goldnead** [comment]
After investigating the response array of the suggestions request in the backend I found that one Page of my Project didn't have a title. Because the link suggestions heavily rely on the title-variable it simply broke when one of the returned items had `null` as the title. (in the response object it is named `text`)
It would be nice if this issue would be resolved in a future update by filtering out each page that has no title or sth. similar. It seems that Statamic can't rely on the user that each page has to have a title.

**Comment by andyra** [comment]
😃 I've been stumped by this one for literally two years. Turns out there were two posts that didn't have titles (and as such didn't show up in the control panel UI), which was causing the Bard link picker to break. You're a gem, @goldnead.
